packages/dev/core/src/Rendering/fluidRenderer/fluidRenderer.ts

```diff
@@ -0,0 +1,547 @@
+
+        for (let i = 0; i < this._targetRenderers.length; ++i) {
+            const targetRenderer = this._targetRenderers[i];
+
+            targetRenderer._initialize();
+
+            if (targetRenderer.camera && targetRenderer._renderPostProcess) {
+                let list = cameras.get(targetRenderer.camera);
+                if (!list) {
+                    list = [[], {}];
+                    cameras.set(targetRenderer.camera, list);
+                }
+                list[0].push(targetRenderer);
+                targetRenderer.camera.attachPostProcess(targetRenderer._renderPostProcess, i);
+            }
+        }
+
+        cameras.forEach((list, camera) => {
+            const firstPostProcess = camera._getFirstPostProcess();
+            if (!firstPostProcess) {
+                return;
+            }
+
+            const [targetRenderers, copyDepthTextures] = list;
+
+            firstPostProcess.onSizeChangedObservable.add(() => {
+                if (!firstPostProcess.inputTexture.depthStencilTexture) {
+                    firstPostProcess.inputTexture.createDepthStencilTexture(0, true, this._engine.isStencilEnable, targetRenderers[0].samples);
+                }
+                for (const targetRenderer of targetRenderers) {
+                    const thicknessRT = targetRenderer._thicknessRenderTarget?.renderTarget;
+                    const thicknessTexture = thicknessRT?.texture;
+                    if (thicknessRT && thicknessTexture) {
+                        const key = thicknessTexture.width + "_" + thicknessTexture.height;
+                        let copyDepthTexture = copyDepthTextures[key];
+                        if (!copyDepthTexture) {
+                            copyDepthTexture = copyDepthTextures[key] = new FluidRenderingDepthTextureCopy(this._engine, thicknessTexture.width, thicknessTexture.height);
+                        }
+                        copyDepthTexture.depthRTWrapper._shareDepth(thicknessRT);
+                    }
+                }
+            });
+        });
+
+        // Dispose the CopyDepthTexture instances that we don't need anymore
+        this._cameras.forEach((list, camera) => {
+            const copyDepthTextures = list[1];
+
+            const list2 = cameras.get(camera);
+            if (!list2) {
+                for (const key in copyDepthTextures) {
+                    copyDepthTextures[key].dispose();
+                }
+            } else {
+                for (const key in copyDepthTextures) {
+                    if (!list2[1][key]) {
+                        copyDepthTextures[key].dispose();
+                    }
+                }
+            }
+        });
+
+        this._cameras.clear();
+        this._cameras = cameras;
+
+        this._setParticleSizeForRenderTargets();
+    }
+
+    private _setParticleSizeForRenderTargets(): void {
+        const particleSizes = new Map<FluidRenderingTargetRenderer, number>();

```

Comment:
That's what I'm doing and that's the purpose of the second loop.

But a user can associate multiple render objects with the same target renderer and each object has its own particle size. However, the target renderer can only handle one particle size, so I have to calculate a single value from all the render object particle sizes: I chose to use the maximum of the values. If this is not what the user wants, they can rework the distribution between render objects and target renderers in their app or even add additional target renderers so that a single render object is associated with a single target renderer.

Then, in the second loop, I simply set the calculated value for each target renderer.